components/twitter/user-tweets.js

```diff
@@ -0,0 +1,96 @@
+const twitter = require('https://github.com/PipedreamHQ/pipedream/components/twitter/twitter.app.js')
+const _ = require('lodash')
+const axios = require('axios')
+const moment = require('moment')
+
+module.exports = {
+  name: "user-tweets", 
+  version: "0.0.1",
+  props: { 
+    db: "$.service.db",
+    twitter,
+    from: { propDefinition: [twitter, "from"] }, 
+    q: { propDefinition: [twitter, "keyword_filter"] },
+    result_type: { propDefinition: [twitter, "result_type"] },
+    count: { propDefinition: [twitter, "count"] },
+    includeRetweets: { propDefinition: [twitter, "includeRetweets"] },
+    includeReplies: { propDefinition: [twitter, "includeReplies"] },
+    enrichTweets: { propDefinition: [twitter, "enrichTweets"] },
+    lang: { propDefinition: [twitter, "lang"] },
+    locale: { propDefinition: [twitter, "locale"] },
+    geocode: { propDefinition: [twitter, "geocode"] },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  async run(event) {
+    const from = `from:${this.from.replace('@','')}`
+    
+    let q = ''
+
+    if(this.q) {
+      q = this.q
+    }
+    
+    const since_id = this.db.get("since_id") || 0
```

Comment:
this is basically "greatest" dedupe strategy -- right? i guess it just needs to pass you the checkpoint so you don't need db?